### PR TITLE
fix flake8 issues F841,E101,E111

### PR DESCRIPTION
**Pull request by gruebel**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

- fixed some flake8 issues
- migrated old tests to the better syntax

### CODE REVIEW

File: checkov/terraform/parser.py [hunk 1]
```diff
@@ -292,7 +292,7 @@ def _internal_dir_load(self, directory: str,
         #                      because it break encapsulations and I don't want to cause confusion about what data
         #                      set it being processed. To avoid this, here's a Callable that will get the data
         #                      map for a particular module reference. (Might be OCD, but...)
-        module_data_retrieval = lambda module_ref: self.out_definitions.get(module_ref)
+        module_data_retrieval = lambda module_ref: self.out_definitions.get(module_ref)  # noqa
```
Comment (nimrodkor) [inline checkov/terraform/parser.py:295]: @gruebel what's wrong with this line that it required `# noqa`?

Comment (gruebel) [reply checkov/terraform/parser.py:295]: the `module_data_retrieval` variable is unused, was not sure, if I can just delete the whole line 😄

Comment (nimrodkor) [reply checkov/terraform/parser.py:295]: If checks pass, it's good to go 🙏

Comment (gruebel) [reply checkov/terraform/parser.py]: removed it 😄

**Pull request merged**